app/containers/System/containers/Firmware/index.js

```diff
@@ -1,7 +1,178 @@
+      variables: {
+        firmwareTrackId,
+        firmwareVersionId,
+      },
+    })
+      .then(() => {
+        refetchAssignmentData();
+        notification.success({
+          message: 'Success',
+          description: 'Track Assignment successfully deleted.',
+        });
+      })
+      .catch(() =>
+        notification.error({
+          message: 'Error',
+          description: 'Track Assignment could not be deleted.',
+        })
+      );
+  };
+  const handleCreateFirmware = (
+    modelId,
+    versionName,
+    description,
+    filename,
+    commit,
+    releaseDate,
+    validationCode
+  ) => {
+    createFirmware({
+      variables: {
+        modelId,
+        versionName,
+        description,
+        filename,
+        commit,
+        releaseDate,
+        validationCode,
+      },
+    })
+      .then(() => {
+        refetch();
+        notification.success({
+          message: 'Success',
+          description: 'Firmware version successfully created.',
+        });
+      })
+      .catch(() =>
+        notification.error({
+          message: 'Error',
+          description: 'Firmware version could not be created.',
+        })
+      );
+  };
+
+  const handleUpdateFirmware = (
+    id,
+    modelId,
+    versionName,
+    description,
+    filename,
+    commit,
+    releaseDate,
+    validationCode,
+    createdTimestamp,
+    lastModifiedTimestamp
+  ) => {
+    updateFirmware({
+      variables: {
+        id,
+        modelId,
+        versionName,
+        description,
+        filename,
+        commit,
+        releaseDate,
+        validationCode,
+        createdTimestamp,
+        lastModifiedTimestamp,
+      },
+    })
+      .then(() => {
+        refetch();
+        notification.success({
+          message: 'Success',
+          description: 'Firmware version successfully updated.',
+        });
+      })
+      .catch(() =>
+        notification.error({
+          message: 'Error',
+          description: 'Firmware version could not be updated.',
+        })
+      );
+  };
+
+  const handleDeleteFirmware = id => {
+    deleteFirmware({
+      variables: {
+        id,
+      },
+    })
+      .then(() => {
+        refetch();
+        notification.success({
+          message: 'Success',
+          description: 'Firmware version successfully deleted.',
+        });
+      })
+      .catch(() =>
+        notification.error({
+          message: 'Error',
+          description: 'Firmware version could not be deleted.',
+        })
+      );
+  };
+
+  if (error) {
+    return (
+      <Alert message="Error" description="Failed to load Firmware data." type="error" showIcon />
+    );
+  }
+
+  if (trackAssignmentError) {
+    return (
+      <Alert
+        message="Error"
+        description="Failed to load Firmware Track Assignment data."
+        type="error"
+        showIcon
+      />
+    );
+  }
+
+  if (loading || trackAssignmentLoading) {
+    return <Loading />;
+  }
```

Comment:
move the errors and loading to ui-library so the two tables can load independently